src/meta/src/hummock/compaction/level_selector.rs

```diff
@@ -284,6 +365,34 @@ impl LevelSelector for DynamicLevelSelector {
             .unwrap_or(false)
     }
 
+    fn waiting_schedule_compaction_bytes(
+        &self,
+        levels: &Levels,
+        level_handlers: &[LevelHandler],
+    ) -> u64 {
+        let ctx = self.inner.calculate_level_base_size(levels);
+        let mut pending_compaction_bytes = 0;
+        pending_compaction_bytes += self.calculate_l0_overlap(
+            levels.l0.as_ref().unwrap(),
+            &levels.levels[ctx.base_level],
+            level_handlers,
+        );
+        for level in &levels.levels {
+            let level_idx = level.level_idx as usize;
+            // The data of last level would not be compact to other level.
+            if level_idx < ctx.base_level || level_idx >= levels.levels.len() {
+                continue;
+            }
+            pending_compaction_bytes += self.calculate_base_level_overlap(
+                ctx.level_max_bytes[level_idx],
+                level,
+                &levels.levels[level_idx],
```

Comment:
use `levels.get_level(level_idx + 1)` to avoid confusion.

Reply:
Fixed